source/extensions/common/aws/signer.h

```diff
@@ -26,6 +26,12 @@ class Signer {
    * @throws EnvoyException if the request cannot be signed.
    */
   virtual void sign(Http::HeaderMap& headers) PURE;
+
+  /**
+   * Temporary for testing...
+   * figure out a better solution before PR
+   */
+  virtual void sign(Http::HeaderMap& headers, const std::string& content_hash) PURE;
```

Comment:
thoughts???

Reply:
Could we just have one function and have content_hash have a default value of an empty string?

Reply:
default arguments are evil though. One way door.